src/mbm.jl

```diff
@@ -0,0 +1,329 @@
+    method::MBM
+) where {T, S <: _MOI.Nonpositives, R}
+    return maximum(_maximize_M(model, ScalarConstraint(objective.func[i], MOI.LessThan(0.0)), constraints, method) for i in 1:objective.set.dimension)
+end
+
+function _maximize_M(
+    model::JuMP.AbstractModel, 
+    objective::VectorConstraint{T, S, R}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T, S <: _MOI.Nonnegatives, R}
+    return maximum(_maximize_M(model, ScalarConstraint(objective.func[i], MOI.GreaterThan(0.0)), constraints, method) for i in 1:objective.set.dimension)
+end
+
+function _maximize_M(
+    model::JuMP.AbstractModel, 
+    objective::VectorConstraint{T, S, R}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T, S <: _MOI.Zeros, R}
+    return max(
+        maximum(_maximize_M(model, ScalarConstraint(objective.func[i], MOI.GreaterThan(0.0)), constraints, method) for i in 1:objective.set.dimension),
+        maximum(_maximize_M(model, ScalarConstraint(objective.func[i], MOI.LessThan(0.0)), constraints, method) for i in 1:objective.set.dimension)
+    )
+end
+
+function _maximize_M(
+    model::JuMP.AbstractModel, 
+    objective::ScalarConstraint{T, S}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T, S <: Union{_MOI.LessThan, _MOI.GreaterThan}}
+    return _mini_model(model, objective, constraints, method)
+end
+
+function _maximize_M(
+    model::JuMP.AbstractModel, 
+    objective::ScalarConstraint{T, S}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T, S <: _MOI.Interval}
+    return max(_mini_model(model, ScalarConstraint(objective.func, MOI.GreaterThan(objective.set.lower)), constraints, method),
+    _mini_model(model, ScalarConstraint(objective.func, MOI.LessThan(objective.set.upper)), constraints, method))
+end
+
+function _maximize_M(
+    model::JuMP.AbstractModel, 
+    objective::ScalarConstraint{T, S}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T, S <: _MOI.EqualTo}
+    return max(
+        _mini_model(model, ScalarConstraint(objective.func, MOI.GreaterThan(objective.set.value)), constraints, method),
+        _mini_model(model, ScalarConstraint(objective.func, MOI.LessThan(objective.set.value)), constraints, method)
+    )
+end
+
+function _maximize_M(
+    ::JuMP.AbstractModel, 
+    ::F, 
+    ::Vector{DisjunctConstraintRef}, 
+    ::MBM
+) where {F}
+    error("This type of constraints and objective constraint has not been implemented for MBM subproblems\nF: $(F)")
+end
+
+#Solve a mini-model to find the maximum value of the objective function for M value
+function _mini_model(
+    model::JuMP.AbstractModel, 
+    objective::ScalarConstraint{T,S}, 
+    constraints::Vector{DisjunctConstraintRef}, 
+    method::MBM
+) where {T,S <: Union{_MOI.LessThan, _MOI.GreaterThan}}
+    sub_model = Model()
+    new_vars = Dict{VariableRef, VariableRef}()
```

Comment:
This will be potentially problematic if we try to use this with the InfiniteOpt extension. Since it doesn't use `JuMP.Model` or `JuMP.VariableRef`. Also stylistically, please preface all JuMP methods/types with `JuMP.`.